### DISCUSSION TITLE
Removing extended description in commit message.

### DISCUSSION THREAD

**Discussion by nandakumar131** [apache/fory#1245] [General]
Can we not include the extended description which brings in additional noise to the commit message while merging the PR?
This makes it hard to navigate or find something in the `git log`.

**Comment by nandakumar131** [comment]
@chaokunyang @wangweipeng2

**Comment by chaokunyang** [comment]
Hi @nandakumar131 We didn't include extended description into commit message, we included commit details into commit message. How about changing it to extended description. This is what apache arrow and ray does. Most extended description of PR are contains linked issue only and a simple description, which won't bring much noise, but give enough details about what this PR do

**Reply by nandakumar131** [reply]
> Most extended description of PR are contains linked issue only and a simple description, which won't bring much noise, but give enough details about what this PR do

While doing squash and merge of the PR the extend description text box in GitHub is auto-populated with all the commit messages that are part of the PR, which can include messages like "fixing checkstyle", "addressing review comments" or repeating the same commit message again. These messages are not very useful in some cases.

While merging the PR, can we quickly check the description to make sure that they add value to the commit message?

I agree that this will put an additional load on the committers, but it will make the `git log` better.

**Reply by chaokunyang** [reply]
Commit details will be populated when `pull request title and commit details` are configured:
![image](https://github.com/alipay/fury/assets/12445254/4beb4031-6cb3-41e1-aa36-a2300e495c1e)

**Reply by nandakumar131** [reply]
Is it possible to edit the "commit details"/"description" while merging?

**Comment by caicancai** [comment]
@nandakumar131 It seems that you want developers to perform operations such as squash and rebase main when submitting. This is what the calcite community is currently trying.
cc @chaokunyang
